tests/ui/implicit_hasher.stderr

```diff
@@ -0,0 +1,184 @@
+error: impl for `HashMap` should be generarized over different hashers
+  --> $DIR/implicit_hasher.rs:11:35
+   |
+11 | impl<K: Hash + Eq, V> Foo<i8> for HashMap<K, V> {
+   |                                   ^^^^^^^^^^^^^
+   |
+   = note: `-D implicit-hasher` implied by `-D warnings`
+help: consider adding a type parameter
+   |
+11 | impl<K: Hash + Eq, V, S: ::std::hash::BuildHasher + Default> Foo<i8> for HashMap<K, V> {
+   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+help: ...and change the type to
```

Comment:
This is still made up of two separate suggestions. I think it'll be less confusing if it were also merged like you did with the constructors.